InnerEye/ML/utils/checkpoint_handling.py

```diff
@@ -252,25 +257,25 @@ def download_checkpoints_to_temp_folder(run: Optional[Run] = None, workspace: Op
     :return: The path to which the files were downloaded. The files are located in that folder, without any further
         subfolders.
     """
+    if not is_running_in_azure_ml() and run is None:
+        raise ValueError("When running outside AzureML, the run to download from must be set.")
     run = run or RUN_CONTEXT
-    # Downloads should go to a temporary folder because downloading the files to the checkpoint folder might
-    # cause artifact conflicts later.
     temp_folder = Path(tempfile.mkdtemp())
-    checkpoint_prefix = f"{DEFAULT_AML_UPLOAD_DIR}/{CHECKPOINT_FOLDER}/"
-    existing_checkpoints = get_run_file_names(run, prefix=checkpoint_prefix)
+    cleaned_prefix = folder.strip("/") + "/"
+    existing_checkpoints = get_run_file_names(run, prefix=cleaned_prefix)
```

Comment:
We still refer to checkpoints here and a few times below, but these files don't necessarily have to be checkpoints. Rename to something more generic?